x-pack/metricbeat/module/statsd/server/data.go

```diff
@@ -111,13 +110,27 @@ func (p *metricProcessor) processSingle(m statsdMetric) error {
 		return nil
 	}
 
+	// parse sample rate. Only applicable for timers and counters
+	var sampleRate float64
+	if m.sampleRate == "" {
+		sampleRate = 1.0
+	} else {
+		var err error
+		sampleRate, err = strconv.ParseFloat(m.sampleRate, 64)
+		if err != nil {
+			return errors.Wrapf(err, "failed to process counter `%s` sample rate `%s`", m.name, m.sampleRate)
+		}
```

Comment:
Just in case, for safety we should also check that `sampleRate` is not `0.0` to avoid divides by zero.